### PR TITLE
Fix walls being attacked after being given/taken

### PR DESCRIPTION
**Pull request by L-e-x-o-n**
After walls are taken or igven the neutral status wqs reset and they were getting attacked by units in range.

### CODE REVIEW

**Comment by sprunk** [comment]
> After walls were taken or given, their neutral status was reset

Sounds like this is the real issue. Check if the engine or game lua is doing it and fix that. The workaround looks okay for now though.

**Comment by L-e-x-o-n** [comment]
> > After walls were taken or given, their neutral status was reset
> 
> Sounds like this is the real issue. Check if the engine or game lua is doing it and fix that. The workaround looks okay for now though.

I did some searching on lua side and couldn't find anything other than thia widget I looked for where neutral units are set but it could be that it's changed from somewhere else 
Not sure how to check what engine does.

**Comment by WatchTheFort** [comment]
To check if it's an engine issue, disable luarules and luaui and see if it still happens.

**Comment by L-e-x-o-n** [comment]
> To check if it's an engine issue, disable luarules and luaui and see if it still happens.

How can I give stuff without luaui and rules?

**Comment by L-e-x-o-n** [comment]
I will try, thanks

**Comment by L-e-x-o-n** [comment]
I was able to replicate the issue with luaui and luarules disabled. It's an engine issue then?

**Comment by L-e-x-o-n** [comment]
I think I found the cause (could be wrong, never did anything engine related).
This line here seems to disable neutral status on change team, which I am assuming is what happens when an unit is given or taken. https://github.com/beyond-all-reason/spring/blob/f9679c363b17e3391255c18de7be095be6de1c9e/rts/Sim/Units/Unit.cpp#L1487

**Comment by WatchTheFort** [comment]
That makes sense, it's the game that is doing something weird, by making a unit that belongs to the enemy team neutral. In that sense, this will have to be fixed game side.

**Comment by L-e-x-o-n** [comment]
> 

How else can you ignore walls from being fired on?

**Comment by WatchTheFort** [comment]
So originally, walls weren't units, but turned into features when they finished. However, this had some unwanted side-effects, like being immediately visible to everyone and not being able to be self-destructed. Then there is are the disguised turrets, like the Dragon's Maw/Claw, which need to be indistinguishable from regular walls.

It's a whole big mess.

File: luarules/gadgets/unit_objectify.lua [hunk 1]
```diff
@@ -35,6 +35,15 @@ if gadgetHandler:IsSyncedCode() then
         end
     end
 
+    function gadget:UnitGiven(unitID, unitDefID, oldTeam, newTeam)
+        if isObject[unitDefID] and Spring.ValidUnitID(unitID) then
+            Spring.SetUnitStealth(unitID, true)
+            Spring.SetUnitSonarStealth(unitID, true)
+            Spring.SetUnitNeutral(unitID, true)
+            Spring.SetUnitBlocking(unitID, true, true, true, true, true, true, false) -- set as crushable
+        end
```
Comment (WatchTheFort) [inline luarules/gadgets/unit_objectify.lua:44]: This should be extracted to a function called from both `UnitCreated` and `UnitGiven`.

**Comment by KyleAnthonyShepherd** [comment]
https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/luarules/gadgets/unit_objectify.lua

This gadget makes walls neutral, I think. 
So it is a game side thing

**Comment by L-e-x-o-n** [comment]
> https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/luarules/gadgets/unit_objectify.lua
> 
> This gadget makes walls neutral, I think. So it is a game side thing

Yes, if you check the changes proposed by this PR it changes that specific gadget.

File: luarules/gadgets/unit_objectify.lua (hunk 1 above)
Comment (L-e-x-o-n) [reply luarules/gadgets/unit_objectify.lua]: Done

**Pull request merged**